apps/sms/test/marionette/mocks/mock_navigator_moz_mobile_message.js

```diff
@@ -192,12 +232,15 @@ Services.obs.addObserver(function(document) {
             } else {
               cursor.result = thread.messages[currentIndex];
             }
-            cursor.onsuccess.call(cursor);
+
+            window.setTimeout(function nextTick() {
```

Comment:
As I mentioned on IRC, it seems that it works correctly now even with 500~ iterations - do you prefer Promise syntax?